### DISCUSSION TITLE
Some warnings against enums

### DISCUSSION THREAD

**Discussion by janvtBE** [StefanMaron/BusinessCentral.LinterCop#538] [New Rule]
Next piece of code is a bit strange concerning creating an enum:

```AL
enum 58100 "VRA Test Enum VRA"
{
    Extensible = true;
    Caption = 'Test Enum';

    value(0; "")
    {
        Caption = 'Empty';
    }
    value(1; "Not Processed")
    {
        Caption = '';
    }
    value(2; "Processing")
    {
        Caption = 'Processing';
    }
    value(3; "Processed")
    {
        Caption = 'Processed';
    }
    value(4; "Error")
    {
        Caption = 'Error';
    }
}
```

An enum with empty name and a caption filled in, like value 0: this is strange, because you cannot select this in de GUI and you cannot use this in AL Code.

An enum with the name filled in and an empty caption, like value 1: this is also strange, this is not selectable in the GUI, allthough you can handle this in AL Code.

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/74171877/b8975936-6d71-4086-9549-21264cb79061)

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/74171877/882eaf21-567b-45c7-ade7-91ffaec9e23a)

So in case of enum 0 and 1, a warning can be helpful to get your focus on this.

Looking forward to your opinion.

**Comment by rvanbekkum** [comment]
The second example is something we do intentionally, but typically with value 0, e.g.:
```
value(0; Undefined)
{
    Caption = '', Locked = true;
}
```
**BUT** in combination with `, Locked = true`.

Maybe, we could have a warning when `, Locked = true` is missing, because otherwise you can have really weird stuff happening (when in some translations the caption is non-empty, while in others it is empty).

**Reply by janvtBE** [reply]
Isn't this solved with implementation of rule LC0041? https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0041

**Reply by rvanbekkum** [reply]
I guess so. It clashes with LC0045 though, so that's why we have disabled that rule.
See: https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/486

**Reply by Arthurvdv** [reply]
@janvtBE, thank you for the suggestion and @rvanbekkum for the additional input on this!

There  two new rules in the (pre)release version of [v0.30.16](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.16) of the LinterCop

- LC0056: Empty Enum values should not have a specified Caption property.
- LC0057: Enum values must have non-empty a Caption to be selectable in the client.

**Reply by janvtBE** [reply]
Thanks @Arthurvdv !
